auth/auth-server-api/src/lib.rs

```diff
@@ -0,0 +1,36 @@
+//! API types for the auth server
+
+#![deny(missing_docs)]
+#![deny(clippy::missing_docs_in_private_items)]
+#![deny(unsafe_code)]
+#![deny(clippy::needless_pass_by_ref_mut)]
+#![feature(trivial_bounds)]
+
+use serde::Deserialize;
+use uuid::Uuid;
+
+// ----------------------
+// | API Key Management |
+// ----------------------
+
+/// The path to create a new API key
+///
+/// POST /api-keys
+pub const API_KEYS_PATH: &str = "api-keys";
+/// The path to mark an API key as inactive
+///
+/// POST /api-keys/{id}/deactivate
+pub const DEACTIVATE_API_KEY_PATH: &str = "deactivate";
```

Comment:
why not make this the entire templated path, i.e. `api-keys/:id/deactivate`? it's confusing in client-side code to import this constant and still have to prepend the first path segments

Reply:
ah, I see how this is useful in `auth-server/main.rs` when using the `warp::path::param`. still, for client-side ergonomics, it may be a good idea to export a separate constant / method that constructs the full / templated path